metric_map.go

```diff
@@ -46,7 +50,10 @@ func (mm *MetricMap) Receive(m *Metric) {
 }
 
 func MergeMaps(mms []*MetricMap) *MetricMap {
-	mm := NewMetricMap()
+	if len(mms) == 0 { // garbage in, garbage out
+		return nil
+	}
+	mm := NewMetricMap(mms[0].Forwarded)
```

Comment:
This is an assumption that all metric maps are forwarded, is that safe to do so?

Reply:
It's an assumption that everything in the slice has the same Forwarded flag, which they should.  We don't do any enforcement of this, because that means we need to propagate an error out, and it's not worth it.

Reply:
Technically we would panic, since an invariant has been broken.